Core/Frameworks/BaikalAdmin/Controller/Install/VersionUpgrade.php

```diff
@@ -146,10 +151,233 @@ protected function upgrade($sVersionFrom, $sVersionTo) {
 			}
 		}
 
-		if(version_compare($sVersionFrom, '0.2.4', '<=')) {
-			# Nothing to do :)
+        if(version_compare($sVersionFrom, '0.3.0', '<')) {
+            // Upgrading from sabre/dav 1.8 schema to 3.1 schema.
+            $pdo = $GLOBALS['DB']->getPDO();
+            if(defined("PROJECT_DB_MYSQL") && PROJECT_DB_MYSQL === TRUE) {
+
+                // MySQL upgrade
+
+                // sabre/dav 2.0 changes
+                foreach(['calendar','addressbook'] as $dataType) {
+
+                    $tableName = $dataType . 's';
+                    $pdo->exec("ALTER TABLE $tableName ADD synctoken INT(11) UNSIGNED NOT NULL DEFAULT '1'");
```

Comment:
PDO should throw an exception if anything fails, but I will double-check to see if we correctly set it to throw exceptions.